### ISSUE TITLE
axe-core throws a warn for treeitem with missing parent while parent is present

### ISSUE THREAD

**Issue by veselints** [dequelabs/axe-core#2897]
Product: 
axe-core

Expectation: 
No warn should be present

Actual: 
Ensures elements with an ARIA role that require parent roles are contained by them. Required ARIA parents role not present: group, tree. The above issue is reported when testing the following markup:
```html
<div tabindex="0" role="tree">
	<li role="treeitem" aria-expanded="false">
		<span>foo</span>
	</li>
	<li role="treeitem" aria-expanded="true">
		<span>
		    bar
			<span></span>
		</span>
		<ul role="group" aria-hidden="false">
			<li role="treeitem">
				<span>ber</span>
			</li>
		</ul>
	</li>
</div>
```

Motivation: 
All `treeitem` elements have either `tree` or `group` as a direct parent.

axe-core version: 4.2.0

**Comment by WilcoFiers** [comment]
@veselints Thanks for reporting. I think axe is tripping up over encountering `treeitem > group > treeitem`, and is expecting that to always be `tree > group > treeitem`. What you've done seems consistent with the ARIA practices, and from some testing I've done this seems to work reasonably well. I'll be sure to put in a fix for this.

**Comment by padmavemulapati** [comment]
Verified with the latest code base - `axe-core-develop -code`, with 4.2.0
Warning seeing because of tripping around `treeitem` and `group`
Addressing `tree`  or `menu` in the parent will resolve the problem
Pass scenarios verified:
`<div role="list"> <div role="listitem"> <div role="group"> <div role="listitem" id="target"></div></div></div></div>`
`  <div role="menu"><div role="group"><p role="listitem" id="target">Nothing here.</p></div></div>`
Fail scenarios:
  `<div role="menu"><div role="menuitem"><div role="group"><div role="menuitem" id="target"></div></div></div></div>`
   ` <div role="list"><div role="listitem"><div role="none"><div role="listitem" id="target"></div></div></div></div>`
Pass results:
  
![image](https://user-images.githubusercontent.com/47409579/120314066-7543fd00-c2f8-11eb-93ba-0c3fa9400e31.png)

Fail results:
![Uploading image.png…]()
